### PR TITLE
Screen AWS markers for Lambda

### PR DESCRIPTION
**Pull request by joe4dev**
Companion PR: https://github.com/localstack/localstack-ext/pull/2159

## Motivation

Fix AWS markers for Lambda according to marker report https://github.com/localstack/localstack/issues/9147

Grepped for some extra occurrences not covered in the marker report.

## Changes

* Set markers to `localstack_only` where applicable
* Remove false positives and rename fixtures or inner asserts to clarify their intent
* Leave some to `unknown` scheduled to be removed at 3.0 (not worth fixing/testing)

### CODE REVIEW

**Comment by dominikschubert** [comment]
> Grepped for some extra occurrences not covered in the marker report.

can you explain what specifically it was that you didn't find in the report? :thinking:

File: tests/aws/services/lambda_/test_lambda_legacy.py [hunk 1]
```diff
@@ -87,8 +89,8 @@ def test_logging_in_local_executor(create_lambda_function, handler_path, aws_cli
 
 
 @pytest.mark.skipif(not is_old_provider(), reason="test does not make valid assertions against AWS")
+@markers.aws.unknown
```
Comment (dominikschubert) [inline tests/aws/services/lambda_/test_lambda_legacy.py:92]: This isn't enforced right now, but please keep the markers at a function level

**Comment by joe4dev** [comment]
> > Grepped for some extra occurrences not covered in the marker report.
> 
> can you explain what specifically it was that you didn't find in the report? 🤔

For example, the false positives were not included:
* `test_image` is a fixture rather than a test and got an `unknown` marker not appearing in the report:  https://github.com/localstack/localstack/pull/9283/files#diff-2d1f022dc5ab81edd6a3789403f602fd45ac88ed37d8151b5b98d1ac617a2ce3L708
* `test_name_and_qualifier` is an inner assertion helper: https://github.com/localstack/localstack/pull/9283/files#diff-2d1f022dc5ab81edd6a3789403f602fd45ac88ed37d8151b5b98d1ac617a2ce3L3381

**Comment by joe4dev** [comment]
> > Grepped for some extra occurrences not covered in the marker report.
> 
> can you explain what specifically it was that you didn't find in the report? 🤔

For example, the false positives were not included:
* `test_image` is a fixture rather than a test and got an `unknown` marker not appearing in the report:  https://github.com/localstack/localstack/pull/9283/files#diff-2d1f022dc5ab81edd6a3789403f602fd45ac88ed37d8151b5b98d1ac617a2ce3L708
* `test_name_and_qualifier` is an inner assertion helper: https://github.com/localstack/localstack/pull/9283/files#diff-2d1f022dc5ab81edd6a3789403f602fd45ac88ed37d8151b5b98d1ac617a2ce3L3381

File: tests/aws/services/lambda_/test_lambda_legacy.py (hunk 1 above)
Comment (joe4dev) [reply tests/aws/services/lambda_/test_lambda_legacy.py:92]: wasn't sure about that as well; will fix it 👍

**Review by dominikschubert — APPROVED** [review]
LGTM. Thanks also for renaming the test_* fixtures :+1:

**Pull request merged**